win32ss/gdi/ntgdi/bitblt.c

```diff
@@ -337,18 +337,22 @@ NtGdiMaskBlt(
     BOOL Status = FALSE;
     EXLATEOBJ exlo;
     XLATEOBJ *XlateObj = NULL;
-    BOOL UsesSource;
+    BOOL UsesSource, UsesPattern;
     ROP4 rop4;
 
     rop4 = WIN32_ROP4_TO_ENG_ROP4(dwRop4);
 
-    UsesSource = ROP4_USES_SOURCE(rop4);
-    if (!hdcDest || (UsesSource && !hdcSrc))
+    if (!hdcDest)
     {
         EngSetLastError(ERROR_INVALID_PARAMETER);
         return FALSE;
     }
 
+    UsesSource = ROP4_USES_SOURCE(rop4);
+    UsesPattern = ROP4_USES_PATTERN(rop4);
+    if (!hdcSrc && (UsesSource || UsesPattern))
+        return FALSE;
```

Comment:
So here, there's no last-error being set?

Reply:
There is no.